### ISSUE TITLE
R 4.1.0 stack overflow

### ISSUE THREAD

**Issue by brainprint** [dirkschumacher/encryptedRmd#30]
Hi,

Thank you for your package.

I just tried it and I got an error:

```
R version 4.1.0 (2021-05-18) -- "Camp Pontanezen"
Copyright (C) 2021 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin17.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

[R.app GUI 1.76 (7976) x86_64-apple-darwin17.0]

> library(encryptedRmd)
> encrypt_html_file(
+   path = "/Users/xxx/Library/Mobile Documents/com~apple~CloudDocs/xxx/xxx/xxx/RMD/DDD.html",
+   output_path = paste0(path, ".enc.html"),
+   key = sodium::random(32L),
+   message_key = TRUE,
+   write_key_file = TRUE,
+   output_template_path = system.file("html-template.html", package = "encryptedRmd"))

Error: segfault from C stack overflow
```

Any clues if it is related to the new R 4.1.0 version? I am using a Mac mini Intel, 11.3.1 macOS. The file to be encrypted is about 10 MB.

Best, RN.

**Comment by dirkschumacher** [comment]
Thanks. Might be related to #22. Can you try the latest GitHub version of {sodium}?

**Comment by dirkschumacher** [comment]
Should be fixed with sodium 1.2
